src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,7 +1,71 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { getMovie } from '../../api';
 import './FindMovie.scss';
+import { MovieCard } from '../MovieCard';
+import { MovieData } from '../../types/MovieData';
+import { movieDataToMovie } from '../../utils/movieDatatoMovie';
+import { ResponseError } from '../../types/ReponseError';
+import { Movie } from '../../types/Movie';
+
+type Props = {
+  setMovies: React.Dispatch<React.SetStateAction<Movie[]>>;
+  checkAlreadyExists: (imdbId: string) => boolean;
+};
+
+export const FindMovie: React.FC<Props> = ({
+  setMovies,
+  checkAlreadyExists,
+}) => {
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<MovieData | null>();
+  const [loading, setLoading] = useState(false);
+  const [isQueryAbsent, setIsQueryAbsent] = useState(false);
+
+  function isMovie(
+    object: MovieData | ResponseError,
+  ): object is MovieData {
+    return (object as MovieData).imdbID !== undefined;
+  }
+
+  const handleAddMovie = () => {
+    if (movie && !checkAlreadyExists(movie.imdbID)) {
+      setMovies(prev => [...prev, movieDataToMovie(movie)]);
+    }
+
+    setMovie(null);
+    setQuery('');
+  };
+
+  const handleQueryChange = (text: string) => {
+    setIsQueryAbsent(false);
+    setQuery(text);
+  };
+
+  const handleMovieRequest = (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+  ) => {
+    e.preventDefault();
+
+    setLoading(true);
+    getMovie(query.trim())
+      .then(result => {
+        if (isMovie(result)) {
+          setMovie({
+            ...result,
+            Poster: result.Poster.startsWith('https')
+              ? result.Poster
+              : 'https://via.placeholder.com/360x270.png?text=no%20preview',
+          });
+        } else {
+          setIsQueryAbsent(true);
+        }
+      })
+      .finally(() => {
+        setLoading(false);
+      });
```

Comment:
the if..else could be declared as a helper 
```
const processMovie = (result) => {
  if (isMovie(result)) {
    setMovie({
      ...result,
      Poster: result.Poster.startsWith('https')
        ? result.Poster
        : 'https://via.placeholder.com/360x270.png?text=no%20preview',
    });
  } else {
    setIsQueryAbsent(true);
  }
}
```

and then
```suggestion
    setLoading(true);
    getMovie(query.trim())
      .then(processMovie)
      //same as `.then(result => callback(result))`
      .finally(() => {
        setLoading(false);
      });
```